### PR TITLE
Document Tier 2 Model Blocker Issues

### PR DESCRIPTION
**Pull request by jeffreyhorn**
## Summary

This PR documents the 6 blocker categories preventing Tier 2 models from parsing, with detailed implementation plans for each.

### High Priority
1. **Dynamic Set Ranges (% syntax)** - Issue #387
   - Blocks 5 models: chain, elec, gasoil, jbearing, polygon
   - 4h effort - Requires preprocessor enhancement for `%variable%` substitution
   
2. **Multi-line Set/Parameter Declarations** - Issue #388
   - Blocks 4 models: gastrans, water, chem, pool
   - 5h effort - Requires grammar updates to handle newlines in declarations

### Medium Priority
3. **Variable/Parameter Attribute Assignment** - Issue #389
   - Blocks 1 model: bearing
   - 3h effort - Requires attribute reference AST nodes

4. **Hyphens in Identifiers** - Issue #390
   - Blocks 1 model: chenery
   - 2h effort - Requires context-sensitive lexing

5. **Variable Indexing in Declarations** - Issue #391
   - Blocks 1 model: inscribedsquare
   - 3h effort - Requires domain specification support

### Low Priority
6. **Multi-line Table Continuation (+)** - Issue #392
   - Blocks 1 model: like
   - 2h effort - Requires table continuation line merging

## Impact

- **Current parse rate:** 5/18 models (27.8%)
- **Potential parse rate:** 18/18 models (100%) after all fixes
- **Total effort estimate:** 19 hours

## Documentation Includes

For each blocker:
- ✅ Detailed problem description with real examples from models
- ✅ Expected behavior and GAMS syntax reference
- ✅ Implementation plan with complexity estimates
- ✅ Grammar/parser changes required
- ✅ Comprehensive test case specifications
- ✅ Success criteria and impact analysis
- ✅ Links to GAMS documentation

## Files Added

- `docs/issues/parser-dynamic-set-ranges.md`
- `docs/issues/parser-multiline-declarations.md`
- `docs/issues/parser-attribute-assignment.md`
- `docs/issues/parser-hyphens-in-identifiers.md`
- `docs/issues/parser-variable-indexing-syntax.md`
- `docs/issues/parser-table-continuation.md`

## Next Steps

These issues provide implementation roadmaps for Sprint 12 and beyond to achieve 100% Tier 2 model parse coverage.

### CODE REVIEW

File: docs/issues/parser-variable-indexing-syntax.md [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+Variables
+    z        "objective value"
+    t(i)     "indexed by set i"
+    x(i,i)   "2-dimensional indexing";
+```
+
+This is equivalent to:
+```gams
+Variables z, t, x;
+# Domain implied by usage
```
[the remark this answers is not in the document]
Comment (jeffreyhorn) [reply docs/issues/parser-variable-indexing-syntax.md]: Fixed in 63663e9. Changed comment from # to * (correct GAMS line comment syntax).

File: docs/issues/parser-variable-indexing-syntax.md [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+Variable x;
+Equation balance;
+balance.. sum(i, x) =e= total;  # Domain inferred from usage
+```
+
+**With explicit domain:**
+```gams
+Variable x(i);
+Equation balance;
+balance.. sum(i, x(i)) =e= total;  # Domain explicit
```
[the remark this answers is not in the document]
Comment (jeffreyhorn) [reply docs/issues/parser-variable-indexing-syntax.md]: Fixed in 63663e9. Changed comments from # to * (correct GAMS line comment syntax) on lines 182 and 189.

File: docs/issues/parser-attribute-assignment.md [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+- [ ] Unit tests for indexed attribute assignments
+- [ ] Integration test with bearing.gms
+- [ ] Edge case tests (invalid attributes, read-only attributes)
+
+## Affected Models
+
+**Tier 2 Models (1 blocked by this issue):**
+- ✅ bearing.gms (154 lines, NLP) - Journal Bearing Problem
+
+**Impact:** Unlocking this model improves Tier 2 parse rate from 27.8% → 33.3% (5/18 → 6/18)
```
[the remark this answers is not in the document]
Comment (jeffreyhorn) [reply docs/issues/parser-attribute-assignment.md]: Fixed in 63663e9. Updated parse rate precision from 27.8% → 33.3% to 27.78% → 33.33% for consistency.

File: docs/issues/parser-attribute-assignment.md [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+x.lo = 0;
+x.up = 100;
+x.l = 50;
+x.scale = 100;
+```
+
+### Test 7: Fixed value
+```gams
+Variable x;
+x.fx = 42;  # Equivalent to x.lo = 42; x.up = 42;
```
[the remark this answers is not in the document]
Comment (jeffreyhorn) [reply docs/issues/parser-attribute-assignment.md]: Fixed in 63663e9. Changed comment from # to * (correct GAMS line comment syntax) on line 298.
